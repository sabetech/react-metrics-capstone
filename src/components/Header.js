import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { setIdle } from '../redux/covid';

const Header = ({
  headerName, navbarTitle, miniTitle, hasBack,
}) => {
  const navigate = useNavigate();
  const dispatch = useDispatch();

  const handleBack = () => {
    dispatch(setIdle());
    navigate(-1);
  };

  return (
    <>
      <nav>
        {hasBack && <button type="button" onClick={() => handleBack()} className="btn-back">â—€</button>}
        <h3 className="heading">{navbarTitle}</h3>
      </nav>
      <div className="sticky-header">
        <div className="card-label">
          <p>{headerName}</p>
        </div>
        <div className="stats-by-country">
          <p>{miniTitle}</p>
        </div>
      </div>
    </>
  );
};

Header.propTypes = {
  headerName: PropTypes.string.isRequired,
  navbarTitle: PropTypes.string.isRequired,
  miniTitle: PropTypes.string.isRequired,
  hasBack: PropTypes.bool.isRequired,
};

export default Header;
