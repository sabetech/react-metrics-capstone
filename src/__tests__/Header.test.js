import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from '../components/Header';
import store from '../redux/configureStore';

describe('Tests: Header', () => {
  it('render test', () => {
    const navBar = render(
      <Provider store={store}>
        <Router>
          <Header />
        </Router>
      </Provider>
      ,
    );
    expect(navBar).toMatchSnapshot();
  });
});
