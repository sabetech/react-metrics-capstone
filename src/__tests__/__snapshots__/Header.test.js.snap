// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests: Header render test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav>
        <h3
          class="heading"
        >
          navTest
        </h3>
      </nav>
      <div
        class="sticky-header"
      >
        <div
          class="card-label"
        >
          <p>
            test
          </p>
        </div>
        <div
          class="stats-by-country"
        >
          <p>
            miniTest
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <nav>
      <h3
        class="heading"
      >
        navTest
      </h3>
    </nav>
    <div
      class="sticky-header"
    >
      <div
        class="card-label"
      >
        <p>
          test
        </p>
      </div>
      <div
        class="stats-by-country"
      >
        <p>
          miniTest
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
